src/lib.rs

```diff
@@ -709,22 +708,22 @@ impl<'s> Parser<'s> {
                         inline::Container::Mark => Container::Mark,
                         inline::Container::InlineLink => Container::Link(
                             match self.inlines.src(inline.span) {
-                                CowStr::Owned(s) => s.replace('\n', "").into(),
                                 s @ CowStr::Borrowed(_) => s,
+                                s => s.replace('\n', "").into(),
                             },
                             LinkType::Span(SpanLinkType::Inline),
                         ),
                         inline::Container::InlineImage => Container::Image(
                             match self.inlines.src(inline.span) {
-                                CowStr::Owned(s) => s.replace('\n', "").into(),
                                 s @ CowStr::Borrowed(_) => s,
+                                s => s.replace('\n', "").into(),
                             },
                             SpanLinkType::Inline,
                         ),
                         inline::Container::ReferenceLink | inline::Container::ReferenceImage => {
                             let tag = match self.inlines.src(inline.span) {
-                                CowStr::Owned(s) => s.replace('\n', " ").into(),
                                 s @ CowStr::Borrowed(_) => s,
+                                s => s.replace('\n', " ").into(),
```

Comment:
would need a custom replace function here if we want to avoid allocation for inline

on the inline_perf branch, this is done only with push, though, the newline is never added to the cowstr.